structured_type_test.go

```diff
@@ -1621,3 +1623,151 @@ func forAllStructureTypeFormats(dbt *DBTest, f func(t *testing.T, format string)
+
+					assertEqualE(t, srtCol.Value(rowIndex), "")
+					isNull := srtCol.IsNull(rowIndex)
+					assertTrueF(t, isNull)
+					record.Release()
+				}
+
+			})
+		}
+	})
+}
+
+func TestSelectingSemistructuredObjectInArrowBatches(t *testing.T) {
+	testcases := []struct {
+		name               string
+		query              string
+		expected           string
+		withUtf8Validation bool
+	}{
+		{
+			name:               "test structured type in arrow batch with utf8 validation, snowflakeType = objectType",
+			withUtf8Validation: true,
+			expected:           `{"s":"someString"}`,
+			query:              "SELECT {'s':'someString'}::OBJECT",
+		},
+		{
+			name:               "test structured type in arrow batch without utf8 validation, snowflakeType = objectType",
+			withUtf8Validation: false,
+			expected:           `{"s":"someString"}`,
+			query:              "SELECT {'s':'someString'}::OBJECT",
+		},
+		{
+			name:               "test structured type in arrow batch without utf8 validation, snowflakeType = arrayType",
+			withUtf8Validation: false,
+			expected:           `[1,2,3]`,
+			query:              "SELECT [1, 2, 3]::ARRAY",
+		},
+		{
+			name:               "test structured type in arrow batch with utf8 validation, snowflakeType = arrayType",
+			withUtf8Validation: true,
+			expected:           `[1,2,3]`,
+			query:              "SELECT [1, 2, 3]::ARRAY",
+		},
+	}
+	skipStructuredTypesTestsOnGHActions(t)
+	runDBTest(t, func(dbt *DBTest) {
+		for _, tc := range testcases {
+			t.Run(tc.name, func(t *testing.T) {
+
+				pool := memory.NewCheckedAllocator(memory.DefaultAllocator)
+				defer pool.AssertSize(t, 0)
+				ctx := WithArrowBatches(WithArrowAllocator(context.Background(), pool))
+				if tc.withUtf8Validation {
+					ctx = WithArrowBatchesUtf8Validation(ctx)
+				}
+
+				var err error
+				var rows driver.Rows
+				err = dbt.conn.Raw(func(sc any) error {
+					queryer, implementsQueryContext := sc.(driver.QueryerContext)
+					assertTrueF(t, implementsQueryContext, "snowflake conn does not implement QueryerContext but needs to")
+					rows, err = queryer.QueryContext(ctx, tc.query, nil)
+					return err
+				})
+				assertNilF(t, err)
+				defer rows.Close()
+
+				sfRows, isSfRows := rows.(SnowflakeRows)
+				assertTrueF(t, isSfRows, "rows are not snowflakeRows")
+
+				batches, err := sfRows.GetArrowBatches()
+				assertNilF(t, err)
+				assertNotEqualF(t, len(batches), 0)
+				batch, err := batches[0].Fetch()
+				assertNilF(t, err)
+				assertNotEqualF(t, len(*batch), 0)
+				for _, record := range *batch {
+					assertEqualF(t, record.NumCols(), int64(1), "unexpected number of columns")
+					assertEqualF(t, record.NumRows(), int64(1), "unexpected number of rows")
+
+					curColIndex := 0
+					rowIndex := 0
+
+					strCol, isStr := record.Column(curColIndex).(*array.String)
+					assertTrueF(t, isStr, "wrong type for column")
+
+					actual := strings.ReplaceAll(strings.ReplaceAll(strCol.Value(rowIndex), " ", ""), "\n", "")
```

Comment:
There is a brand new (from today!) function called `assertEqualIgnoringWhitespace` - may be useful!

Reply:
Oh cool! I will rebase and use it